devito/ir/iet/nodes.py

```diff
@@ -484,6 +484,24 @@ def __init__(self, expr, pragmas=None):
         super().__init__(expr, pragmas=pragmas, operation=OpInc)
 
 
+class ActionExpr(Expression):
+
+    def __init__(self, expr, pragmas=None, operation=OpAction,
+                 target=None, solver_parameters=None):
+        super().__init__(expr, pragmas=pragmas, operation=operation)
+        self.target = target
+        self.solver_parameters = solver_parameters
+
+
+class RHSExpr(Expression):
```

Comment:
this is essentially identically to ActionExpr, so perhaps we could have a common superclass

the default kwarg `operation` could become a subclass property